projects/storefrontlib/shared/components/chat-messaging/avatar/avatar.component.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { I18nTestingModule } from '@spartacus/core';
+import { IconModule } from 'projects/storefrontlib/cms-components';
+import { MessageEvent } from '../messaging';
+import { AvatarComponent } from './avatar.component';
+
+const mockEvent: MessageEvent = {
+  author: 'Mark Rivers',
+  rightAlign: true,
+};
+
+describe('AvatarComponent', () => {
+  let component: AvatarComponent;
+  let fixture: ComponentFixture<AvatarComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [I18nTestingModule, IconModule],
+      declarations: [AvatarComponent],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AvatarComponent);
+    component = fixture.componentInstance;
+    component.message = mockEvent;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should return initials', () => {
+    let result = component.getInitials('Mark Rivers');
+
```

Comment:
If the icon for 'anonmyous' (ghost writer) - missing author,

then add the test case for that

Reply:
sure